sos/plugins/origin.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (C) 2016 Red Hat, Inc., Pep Turro Mauri <pep@redhat.com>
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+from sos.plugins import Plugin, RedHatPlugin
+import os.path
+
+
+# This plugin collects static configuration and runtime information
+# about OpenShift Origin based environments, like Atomic Platform or
+# OpenShift Enterprise 3+.
+
+# Some clarifications on naming.
+# OpenShift Origin is the upstream project for OpenShift Enterprise.
+# However, it actually refers to two different code bases:
+#  * Origin M5 and later (https://github.com/openshift/origin)
+#    which is upstream for OpenShift 3.x and later.
+#    This is what this plugin handles
+#  * Origin M4 and earlier (https://github.com/openshift/origin-server)
+#    which is upstream for OpenShift 1.x and 2.x.
+#    This is handled by the plugin in openshift.py
+# Origin M5 is also the code base behind Atomic Platform,
+# not to be confused with Atomic Host (http://projectatomic.io)
+
+# Note that this plugin should be used in conjunction with other plugins
+# in order to capture relevant data: the Kubernetes plugin for the
+# masters, the Docker plugin for the nodes, and also generic
+# plugins (e.g. for /etc/sysconfig entries, network setup etc)
+class OpenShiftOrigin(Plugin):
+    ''' OpenShift Origin '''
+
+    master_base_dir = "/etc/origin/master"
+    node_base_dir = "/etc/origin/node"
+    master_cfg = os.path.join(master_base_dir, "master-config.yaml")
+    node_cfg = os.path.join(node_base_dir, "node-config.yaml")
+    admin_cfg = os.path.join(master_base_dir, "admin.kubeconfig")
+    oc_cmd = "oc --config=%s" % admin_cfg
```

Comment:
Ack, I'll move the name and profile to the top. I need to keep `files` last though as it references instance variables for paths.